src/types/api.js

```diff
@@ -84,13 +84,17 @@
  */
 
 /**
- * TODO - Fill out remaining properties
- *
  * @typedef Profile
  * @prop {string|null} userid
  * @prop {Object} preferences
- * @prop {boolean} [preferences.show_sidebar_tutorial]
+ *   @prop {boolean} [preferences.show_sidebar_tutorial]
  * @prop {Object.<string, boolean>} features
+ * @prop {Object[]} [groups]
```

Comment:
There appears to be group prop here but its different than our `Group` type. Ac-Hoc prop seems reasonable.

Reply:
The `groups` prop is deprecated and should not be used by the client. I would omit it here and perhaps add a note in the type description that deprecated properties are not included. Alternatively, you could list it separately from the others and use the `unknown` type to enforce that no code uses it:

```js
 * @prop {unknown} [groups] - Deprecated.
```

`unknown` is like `any` but code must check its actual type somehow before using it.